### ISSUE TITLE
Instant Search: Heading hierarchy for accessibility

### ISSUE THREAD

**Issue by jpumfreyamergin** [Automattic/jetpack#22156]
### Impacted plugin

Jetpack

### Steps to Reproduce

Screenshot below in 'What actually happened' or

1. Open Instant Search modal
2. Add search query
2. Note the nested headings of filter types and skipped heading levels 

### A clear and concise description of what you expected to happen.

Ideally, there should be a heading level 1 for the full search dialog box which can be hidden with the `.screen-reader-text` class with results and filters separated by heading level 2.

The current heading level 1, 'Search Results', should be heading level 2 as each subsequent result is a heading level 3. 

'Filter Options' should be heading level 2 with subsequent filters heading level 3.

e.g.
![image](https://user-images.githubusercontent.com/52807006/147011456-b4e44ba3-8f32-427f-9b99-6fa50f51d21a.png)

### What actually happened

Please see below screenshot. The heading structure between results and filters are incorrect with no heading level 2 to separate results and filters.

Filters are nested under the last search result.

The 'Filter Options' section has no heading at all and immediately skips to heading level 4. Therefore, filter headings are nested under the last search result. 

e.g.
![image](https://user-images.githubusercontent.com/52807006/147010331-63786d70-c2ed-4168-a5a5-0b81d2cfbbba.png)

### Operating System

Windows

### OS Version

Any OS

### Browser

Firefox

### Browser Version(s)

Any browser
